packages/agreement-platformstate-writer/src/utils.ts

```diff
@@ -92,3 +111,434 @@ export const deleteAgreementEntry = async (
   const command = new DeleteItemCommand(input);
   await dynamoDBClient.send(command);
 };
+
+export const updateAgreementStateInPlatformStatesEntry = async (
+  dynamoDBClient: DynamoDBClient,
+  primaryKey: PlatformStatesAgreementPK,
+  state: ItemState,
+  version: number
+): Promise<void> => {
+  const input: UpdateItemInput = {
+    ConditionExpression: "attribute_exists(PK)",
+    Key: {
+      PK: {
+        S: primaryKey,
+      },
+    },
+    ExpressionAttributeValues: {
+      ":newState": {
+        S: state,
+      },
+      ":newVersion": {
+        N: version.toString(),
+      },
+      ":newUpdateAt": {
+        S: new Date().toISOString(),
+      },
+    },
+    ExpressionAttributeNames: {
+      "#state": "state",
+    },
+    UpdateExpression:
+      "SET #state = :newState, version = :newVersion, updatedAt = :newUpdateAt",
+    TableName: config.tokenGenerationReadModelTableNamePlatform,
+    ReturnValues: "NONE",
+  };
+  const command = new UpdateItemCommand(input);
+  await dynamoDBClient.send(command);
+};
+
+export const agreementStateToItemState = (state: AgreementState): ItemState =>
+  state === agreementState.active ? itemState.active : itemState.inactive;
+
+export const updateAgreementStateInTokenGenerationStatesTableEntries = async ({
+  entriesToUpdate,
+  agreementState,
+  dynamoDBClient,
+}: {
+  entriesToUpdate: TokenGenerationStatesClientPurposeEntry[];
+  agreementState: AgreementState;
+  dynamoDBClient: DynamoDBClient;
+}): Promise<void> => {
+  for (const entry of entriesToUpdate) {
+    const input: UpdateItemInput = {
+      // ConditionExpression to avoid upsert
+      ConditionExpression: "attribute_exists(GSIPK_eserviceId_descriptorId)",
+      Key: {
+        PK: {
+          S: entry.PK,
+        },
+      },
+      ExpressionAttributeValues: {
+        ":newState": {
+          S: agreementStateToItemState(agreementState),
+        },
+        ":newUpdateAt": {
+          S: new Date().toISOString(),
+        },
+      },
+      UpdateExpression:
+        "SET agreementState = :newState, updatedAt = :newUpdateAt",
+      TableName: config.tokenGenerationReadModelTableNameTokenGeneration,
+      ReturnValues: "NONE",
+    };
+    const command = new UpdateItemCommand(input);
+    await dynamoDBClient.send(command);
+  }
+};
+
+export const updateAgreementStateInTokenGenerationStatesTableEntriesPlusDescriptorInfo =
```

Comment:
```suggestion
export const updateAgreementOnTokenStatesWithDescriptor =
```

Reply:
Renamed to `updateAgreementStateAndDescriptorInfoOnTokenStatesEntries`